Code.gs

```diff
@@ -0,0 +1,611 @@
+  userPropertiesService.setProperty(key, obj);
+}
+
+function setUserProperties(
+  minEmails = defaults.MIN_EMAILS,
+  autoArchive = defaults.AUTO_ARCHIVE,
+  label = defaults.LABEL,
+  onlyPrimary = defaults.ONLY_PRIMARY,
+  autoReply = defaults.AUTO_REPLY,
+  replyText = defaults.REPLY_TEXT
+) {
+
+  let saveObject = {
+    MIN_EMAILS: minEmails.toString(),
+    AUTO_ARCHIVE: autoArchive.toString(),
+    LABEL: label,
+    ONLY_PRIMARY: onlyPrimary.toString(),
+    AUTO_REPLY: autoReply.toString(),
+    REPLY_TEXT: replyText,
+  };
+
+  let userPropertiesService = PropertiesService.getUserProperties();
+  userPropertiesService.setProperties(saveObject);
+}
+
+// Function to get a user property by key. If the key doesn't exist, it sets the default value.
+function getUserProperty(key) {
+  let userProperties = PropertiesService.getUserProperties().getProperties();
+  if (
+    (userProperties[key] === undefined || userProperties[key] === null) &&
+    defaults[key] !== undefined
+  ) {
+    setUserProperty(key, defaults[key]);
+    userProperties = PropertiesService.getUserProperties().getProperties();
+  }
+  return userProperties[key];
+}
+
+// Function to get all user properties and convert them to the correct types
+function getUserProperties() {
+  let userProperties = {};
+  userProperties[MIN_EMAILS] = parseInt(getUserProperty(MIN_EMAILS));
+  userProperties[AUTO_ARCHIVE] = getUserProperty(AUTO_ARCHIVE) === "true";
+  userProperties[LABEL] = getUserProperty(LABEL);
+  userProperties[ONLY_PRIMARY] = getUserProperty(ONLY_PRIMARY) === "true";
+  userProperties[AUTO_REPLY] = getUserProperty(AUTO_REPLY) === "true";
+  userProperties[REPLY_TEXT] = getUserProperty(REPLY_TEXT);
+
+  return userProperties;
+}
+
+/**
+ * The createFilter function generates a filter query for the GmailApp search. 
+ * It uses the Gmail search syntax to find emails in the user's inbox which 
+ * are addressed to the user and are newer than 2 days.
+ * 
+ * If @param {onlyPrimary} parameter is true, the function will only return 
+ * emails from the 'Primary' category of the inbox.
+**/
+function createFilter(onlyPrimary = true) {
+  let inbox_filter = ["to:me", "in:inbox", "newer_than:10d"];
+
+  let filter = inbox_filter.join(" ");
+  if (onlyPrimary) {
+    filter = filter + " category:primary";
+  }
+  return filter;
+}
+
+/** 
+ * processInboxByBlade processes the user's inbox. It finds email threads 
+ * matching the created filter in batches of 5.
+ * If a batch has less than 5 threads, it stops fetching more. 
+ * It then checks each thread for being unsolicited and increments totalMovedCount. 
+ * totalMovedCount, the count of potentially moved threads, is returned.
+**/
+function processInboxByBlade() {
+  const maxThreads = 5;
+  let startIndex = 0;
+
+  let onlyPrimary = getUserProperty(ONLY_PRIMARY) === "true";
+  let searchFilter = createFilter(onlyPrimary);
+  Logger.log('SF: ' + searchFilter);
+  let allThreads = [];
+
+  while (true) {
```

Comment:
Google Apps Scripts imposes a limit on the number of seconds a function can run. This infinite loop can potentially exceed that limit.

Reply:
Done. Limited to 50 threads at max. Tested and it is the upper limit for our processing in 45 secs time limit by Google.